lib/js/components/Modals/Modal/Modal.vue

```diff
@@ -0,0 +1,417 @@
+						class="modal__headerFeatureIcon"
+						:color="calcHeaderFeatureIconColor"
+						:icon="headerFeatureIcon"
+						:size="FEATURE_ICON_SIZES.X_LARGE"
+						:double-background="true"
+					/>
+					<h3
+						class="modal__headerTitle"
+						:class="{ '-small': headerTitleSize === MODAL_HEADER_TITLE_SIZES.SMALL }"
+						>{{ headerTitle }}</h3
+					>
+					<h6 v-if="headerSubtitle" class="modal__headerSubtitle">{{
+						headerSubtitle
+					}}</h6>
+				</div>
+				<div v-if="$slots.default" class="modal__textContent">
+					<slot />
+				</div>
+				<div class="modal__footer" :class="{ '-singleColumn': calcSingleColumn }">
+					<div
+						v-if="footerTertiaryButtonText || footerCheckboxText"
+						class="modal__footerColumn --ctaSecondary"
+					>
+						<div v-if="footerCheckboxText" ref="button" class="modal__checkbox">
+							<input
+								id="modal__checkboxInput"
+								type="checkbox"
+								:checked="false"
+								class="modal__checkboxInput"
+								@change="$emit('checkbox-change')"
+							/>
+							<label for="modal__checkboxInput" class="modal__checkboxLabel">
+								{{ footerCheckboxText }}
+							</label>
+						</div>
+						<wnl-button
+							v-if="footerTertiaryButtonText"
+							ref="button"
+							class="modal__buttonSecondary"
+							:type="BUTTON_TYPES.TEXT"
+							:color="BUTTON_COLORS.NEUTRAL"
+							:icon-left="ICONS[footerTertiaryButtonIcon]"
+						>
+							{{ footerTertiaryButtonText }}
+						</wnl-button>
+					</div>
+					<div
+						v-if="footerSecondaryButtonText || footerPrimaryButtonText"
+						class="modal__footerColumn --cta"
+					>
+						<wnl-button
+							v-if="footerSecondaryButtonText"
+							ref="button"
+							class="modal__buttonSecondary"
+							:type="BUTTON_TYPES.OUTLINED"
+							:color="calcFooterSecondaryButtonColor"
+							:icon-right="ICONS[footerSecondaryButtonIcon]"
+						>
+							{{ footerSecondaryButtonText }}
+						</wnl-button>
+						<wnl-button
+							v-if="footerPrimaryButtonText"
+							ref="button"
+							class="modal__buttonPrimary"
+							:color="calcFooterPrimaryButtonColor"
+							:icon-right="ICONS[footerPrimaryButtonIcon]"
+						>
+							{{ footerPrimaryButtonText }}
+						</wnl-button>
+					</div>
+				</div>
+			</div>
+		</div>
+	</div>
+</template>
+
+<style lang="scss" scoped>
+@import '../../../../styles/settings/spacings';
+@import '../../../../styles/settings/radiuses';
+@import '../../../../styles/settings/shadows';
+@import '../../../../styles/settings/colors/tokens';
+@import '../../../../styles/settings/typography/tokens';
+@import '../../../../styles/settings/icons';
+@import '../../../../styles/settings/media-queries';
+
+$modal-medium-width: 700px;
+$modal-small-width: 460px;
+$image-height: 200px;
+$image-height-small: 140px;
+
+.modal {
+	$self: &;
+
+	background: $color-default-overlay;
+	padding: $space-l $space-s;
+
+	@media #{breakpoint-s()} {
+		padding: $space-l;
+	}
+
+	&__wrapper {
```

Comment:
don't we need to limit height? Especially on small modals?

Reply:
I set 80% of view height, but we should ask UX.

Reply:
did you ask?

Reply:
[`6eecc84` (#248)](https://github.com/bethinkpl/design-system/pull/248/commits/6eecc84c0d0ecffbdae6ad58ea04c583d9e8f54d)